app/assets/javascripts/foundation_extras.js

```diff
@@ -3,26 +3,29 @@
   App.FoundationExtras = {
     clearSticky: function() {
       if ($("[data-sticky]").length) {
-        $("[data-sticky]").foundation("destroy");
+        $("[data-sticky],[data-stick-to]").foundation("destroy");
       }
     },
-    mobile_ui_init: function() {
-      $(window).trigger("load.zf.sticky");
-    },
-    desktop_ui_init: function() {
-      $(window).trigger("init.zf.sticky");
-    },
     initialize: function() {
-      $(document).foundation();
-      $(window).trigger("resize");
-      window.addEventListener("popstate", this.clearSticky, false);
-      $(function() {
-        if ($(window).width() < 620) {
-          App.FoundationExtras.mobile_ui_init();
-        } else {
-          App.FoundationExtras.desktop_ui_init();
+      // Fix sticky elements restored from browser cache before Foundation initialization
+      // Now data-sticky-to property is mandatory to make this patch to work
+      if ($("[data-sticky]").length !== $("[data-stick-to]").length) {
+        $("[data-stick-to]").attr("data-sticky", "true");
+      }
+
+      $("[data-sticky]").each(function(_, sticky) {
+        var breakpoints = $(sticky).data("sticky-on") || "small medium large xlarge xxlarge";
+        breakpoints = breakpoints.split(" ");
+        if (breakpoints.indexOf(Foundation.MediaQuery.current) > -1) {
```

Comment:
Hi @javierm, Do you know how can i solve this kind of Hound violation? I guess i should use some kind of `import`statement to add `Foundation` class to application so linter does not complain 🤔

Reply:
I think you can add it to `globals` in the `.eslintrc` file :thinking:.

Reply:
Thanks @javierm, by running `eslint` tool locally does not complain anymore abount Foundation class after adding it yo the `.eslintrc` as you suggested but i am not sure the reason because Hound is not marking this violation as “Outdated”.

Reply:
I am going to move some commits to another PR to keep this PR with commits related with jQuery update. I guess that this violation will be thrown at new PR again. Lets see ...